src/classes/class-resolver.js

```diff
@@ -0,0 +1,27 @@
+const path = require('path')
+
+const TYPES = Object.freeze({
+  adapter: 'adapters',
+  serializer: 'serializers',
+  mixin: 'mixins',
+  store: 'stores',
+  model: 'models',
+  route: 'routes',
+  initializer: 'initializers',
+  middleware: 'middleware',
+  service: 'services'
+})
+
+module.exports = class ClassResolver {
+  static resolve (type, name) {
+    try {
+      return require(path.join(process.cwd(), 'app', TYPES[type], name))
+    } catch (e) {
+      try {
+        return require(path.join(process.cwd(), 'app', TYPES[type], 'application'))
```

Comment:
See note above about `process.cwd()`.

1/10 out of scope of PR
8/10 in general would recommend just having some helper function/utility that allows you to identify the location of a file in the app. I don't think `process.cwd()` will be sufficient in the end. It's ok to require the user to run commands from the client directory when building ember, because you're preprocessing, but a server is more like to be run automatically.

```js
// ...
return require(projectFile('app', TYPES[type], name))
// ...

function projectFile () {
  let args = convertArgsToArray(arguments)
  args.unshift(reliableWayOfDeterminingProjectRoot())
  return path.join(args)
}
```